test/cppsuite/test_harness/runtime_monitor.cxx

```diff
@@ -237,56 +252,98 @@ runtime_monitor::get_stat(scoped_cursor &cursor, int stat_field, int64_t *valuep
 }
 
 runtime_monitor::runtime_monitor(configuration *config, database &database)
-    : component("runtime_monitor", config), _postrun_stats(config), _database(database)
+    : component("runtime_monitor", config), _database(database)
 {
 }
 
 runtime_monitor::~runtime_monitor()
 {
-    for (auto &it : _stats)
-        delete it;
+    for (auto &stat : _stats)
+        delete stat;
     _stats.clear();
 }
 
 void
 runtime_monitor::load()
 {
-    configuration *sub_config;
-    std::string statistic_list;
-
     /* Load the general component things. */
     component::load();
 
+    /* If the component is enabled, load all the known statistics. */
     if (_enabled) {
-        _session = connection_manager::instance().create_session();
 
-        /* Open our statistic cursor. */
-        _cursor = _session.open_scoped_cursor(STATISTICS_URI);
+        configuration *stat_config = _config->get_subconfig(STAT_CACHE_SIZE);
+        _stats.push_back(new cache_limit_statistic(stat_config, STAT_CACHE_SIZE));
+        delete stat_config;
```

Comment:
Maybe we should take advantage of the fact that this is C++ and use `std::unique_ptr` instead of deleting manually?

Reply:
I agree with @AlexeyAnisimovAU 's comment - we should be aiming to use smart pointers everywhere there's pointer ownership (such as this).

Reply:
That's a very good idea, I have added smart pointers in aa3cd46e1c8710696e342810c848060498ad6511. Let me know if that's what you had in mind.